business/istio_validations.go

```diff
@@ -480,6 +473,62 @@ func (in *IstioValidationsService) fetchDetails(rValue *kubernetes.IstioDetails,
 	}
 }
 
+func (in *IstioValidationsService) fetchAllVirtualServices(istioDetails *kubernetes.IstioDetails, namespace string, errChan chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	if len(errChan) == 0 {
+		nss, err := in.businessLayer.Namespace.GetNamespaces()
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		currentVSs, err := in.fetchVirtualServices(namespace)
+		if err != nil {
+			select {
+			case errChan <- err:
+			default:
+			}
+		} else {
+			for _, ns := range nss {
+				vsList, err := in.fetchVirtualServices(ns.Name)
+				if err != nil {
```

Comment:
In go it is pretty common to avoid huge indentation by applying `early returns`. So I'd encourage you to use the `continue` statement and remove the `else` clause.

Plus, I think you don't need to use the `select` clause to send the errors to a channel.